test/e2e/test_e2e.py

```diff
@@ -69,10 +70,10 @@ def run_program(
     quantum_processor_id: str,
     client_configuration: QCSClientConfiguration,
     engagement_manager: EngagementManager,
-):
+)-> np.ndarray:
     qc = get_qc(
         quantum_processor_id,
         client_configuration=client_configuration,
         engagement_manager=engagement_manager,
     )
-    return qc.run(qc.compile(program))
+    return qc.run(qc.compile(program)).readout_data['ro']
```

Comment:
You'll want this change under `test_basic_program` too. You can run these against a QVM for a sanity check with `make e2e TEST_QUANTUM_PROCESSOR=2q-qvm`

Reply:
Bit of a problem here. When I try running these (after correcting by adding `.readout_data["ro"]`), I get issues due to thread locks... :'(

```
TypeError: can't pickle _thread.lock objects
```

Investigating...

Reply:
I'll try moving the qc to outside of the threads, since they're safer there now

Reply:
OK, fixed for you. I just wanted my name on this PR's commit history ;P

Reply:
you deserve it! thanks for pushing that fix